src/js/components/TextInput/TextInput.js

```diff
@@ -119,6 +119,26 @@ const TextInput = forwardRef(
     const [focus, setFocus] = useState();
     const [showDrop, setShowDrop] = useState();
 
+    const handleTextSelect = useRef();
+    const handleSuggestionSelect = useRef();
+
+    // assigns select handlers
+    useEffect(() => {
```

Comment:
Would memo hooks be simpler than ref hooks?
```
const handleSuggestionSelect = useMemo(onSelect && !onSuggestionSelect ? onSelect : onSuggestionSelect, [onSelect, onSuggestionsSelect]);
const handleTextSelect = useMemo(onSelect && onSuggestionSelect ? onSelect : undefined, [onSelect, onSuggestionsSelect]);
```
just wondering

Reply:
changes made